cms/models.py

```diff
@@ -744,6 +771,98 @@ def get_context(self, request, *args, **kwargs):
         }
 
 
+class ExternalCoursePage(ProductPage):
+    """
+    CMS page representing an external course
+    """
+
+    template = "product_page.html"
+
+    parent_page_types = ["CourseIndexPage"]
+
+    external_url = models.URLField(
+        null=False, blank=False, help_text="The URL of the external course web page."
+    )
+    readable_id = models.CharField(
+        max_length=64,
+        null=False,
+        blank=False,
+        unique=True,
+        help_text="The readable ID of the external course. Appears in URL, has to be unique.",
+    )
+    start_date = models.DateField(
+        null=True, blank=True, help_text="The start date of the external course."
+    )
```

Comment:
we don't have end_date here, so how we will decide the visibility of course on catalog page?

Reply:
The same as other pages, with start_date, title sorting. As per my discussion with @pdpinch marketing will take care of keeping this updated.